presto-main/src/main/java/com/facebook/presto/execution/QueryStats.java

```diff
@@ -261,6 +269,8 @@ public QueryStats(
         this.stageGcStatistics = ImmutableList.copyOf(requireNonNull(stageGcStatistics, "stageGcStatistics is null"));
 
         this.operatorSummaries = ImmutableList.copyOf(requireNonNull(operatorSummaries, "operatorSummaries is null"));
+
+        this.runtimeStats = runtimeStats;
```

Comment:
requireNonNull

Reply:
This could be null if the QueryStats is deserialized from a json generated by an older binary version. So this may cause a binary incompatibility issue. Maybe it is better to allow this to be null. What do you think?

Reply:
Updated the logic to create a new object if it is null.